File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
	static int V;
	static List<Integer>[] graph;
	static StringBuilder sb;

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		sb = new StringBuilder();
		
		StringTokenizer st = new StringTokenizer(br.readLine(), " ");
		V = Integer.parseInt(st.nextToken());
		int E = Integer.parseInt(st.nextToken());
		int start = Integer.parseInt(st.nextToken()) - 1;
		
		graph = new ArrayList[V];
		for(int i=0; i<V; i++) {
			graph[i] = new ArrayList<>();
		}
		
		for(int i=0; i<E; i++) {
			st = new StringTokenizer(br.readLine(), " ");
			int s = Integer.parseInt(st.nextToken()) - 1;
			int e = Integer.parseInt(st.nextToken()) - 1;
			
			graph[s].add(e);
			graph[e].add(s);
		}
		
		for(int i=0; i<V; i++) {
			Collections.sort(graph[i]);
		}
		
		dfs(start);
		bfs(start);
		
		bw.write(sb.toString());
		bw.flush(); bw.close(); br.close();
	}
	
	static void dfs(int start) {
		boolean[] visited = new boolean[V];
		visited[start] = true;
		dfs_recur(start, 0, visited);
		sb.append("\n");
	}
	
	static void dfs_recur(int current, int cnt, boolean[] visited) {
		if(cnt == V) {
			sb.append("\n");
			return;
		};
		
		sb.append(current+1).append(" ");
		
		for(int vertex : graph[current]) {
			if(!visited[vertex]) {
				visited[vertex] = true;
				dfs_recur(vertex, cnt+1, visited);
			}
		}
	}

	static void bfs(int start) {
		Queue<Integer> queue = new ArrayDeque<Integer>();
		boolean[] visited = new boolean[V];
		
		queue.offer(start);
		visited[start] = true;
		
		while(!queue.isEmpty()) {
			int current = queue.poll();
			sb.append(current+1).append(" ");
			for(int vertex : graph[current]) {
				if(!visited[vertex]) {
					visited[vertex] = true;
					queue.offer(vertex);
				}
			}
		}
		sb.append("\n");
	}
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
def DFS(matrix, start):
    visited[start] = 1
    print(start, end= " ")
    for i in range(1,N+1):
        if matrix[start][i] == 1 and visited[i] == 0:
            DFS(matrix, i)


def BFS(matrix, start):
    queue = [start]
    visited = [0]*(N+1)
    visited[start] = 1

    while queue:
        node = queue.pop(0)
        print(node, end = " ")
        for i in range(N+1):
            if visited[i] == 0 and matrix[node][i] == 1:
                queue.append(i)
                visited[i] = 1
    

N, M, s = map(int, input().split())
matrix = [[0]*(N+1) for i in range(N+1)]

for i in range(M):
    start, end = map(int, input().split())
    matrix[start][end], matrix[end][start] = 1, 1

visited = [0]*(N+1)

DFS(matrix, s)
print()
BFS(matrix, s)